### PR TITLE
Implement getIncidentDocuments for Elasticsearch

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

This PR implements `getIncidentDocuments` for the `ElasticsearchIncidentUpdateRepository`.

## Related issues

related to #24930

### CODE REVIEW

**Comment by npepinpe** [comment]
Blocked by #25144

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/incident/ElasticsearchIncidentUpdateRepository.java
Comment (npepinpe) [inline zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/incident/ElasticsearchIncidentUpdateRepository.java:80]: :thought_balloon: This may not be really necessary, because the partition ID is actually encoded in the key/ID - but this relies on the fact that ID == key, which I'm not super keen on. So I added the filter here.

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/incident/ElasticsearchIncidentUpdateRepository.java
Comment (npepinpe) [inline zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/incident/ElasticsearchIncidentUpdateRepository.java:87]: :exclamation: I removed the sort by key from the original implementation. I tried to understand why it was there in the first place, and honestly it doesn't make much sense. I assume it's a historical leftover?

Sorting by key (since keys are monotonically generated) gives you _some_ indication of chronology - but I still don't see how it's relevant here in any way, or how it used to be before that.

Comment (entangled90) [reply zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/incident/ElasticsearchIncidentUpdateRepository.java:87]: :thought_balloon:  Should we remove the sorting after we tested that this implementation is working as expected? This may help simplify testing it since we have done less changes. 
Removing the sorting afterwards is quite quick.

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/incident/ElasticsearchIncidentUpdateRepository.java
Comment (entangled90) [inline zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/incident/ElasticsearchIncidentUpdateRepository.java:128]: :question: is there any difference in using `value` vs `anyValue`?

**Review by entangled90 — APPROVED** [review]
Look good. Personally I would keep the sorting for the moment but I think it's your call :+1:

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/incident/ElasticsearchIncidentUpdateRepository.java
Comment (npepinpe) [reply zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/incident/ElasticsearchIncidentUpdateRepository.java:88]: Added the sort back just in case, you're right, easy to remove it later. I don't know if it has any sort of performance implications on the ES side (probably - but then again, unclear if it's important).

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/incident/ElasticsearchIncidentUpdateRepository.java
Comment (npepinpe) [reply zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/incident/ElasticsearchIncidentUpdateRepository.java:129]: AFAIK it's just a short cut, but I didn't dig too much into it. I mostly lifted it from the old code anyway, as I wasn't sure how the client will serialize an enum.

**Pull request merged**
